### PR TITLE
Improve vector/list type when push_back with rvalue

### PR DESCRIPTION
**Pull request by xuzifu666**
### Changelog category (leave one):
- New Feature
- Improvement
- Performance Improvement
- Backward Incompatible Change
- Build/Testing/Packaging Improvement
- Documentation (changelog entry is not required)
- Bug Fix (user-visible misbehavior in an official stable release)
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by xuzifu666** [comment]
@tavplubix hi，could you have a review plz？

File: src/Core/Block.cpp [hunk 1]
```diff
@@ -631,7 +631,7 @@ Names Block::getNames() const
     res.reserve(columns());
 
     for (const auto & elem : data)
-        res.push_back(elem.name);
+        res.push_back(std::move(elem.name));
```
Comment (alexey-milovidov) [inline src/Core/Block.cpp:634]: This looks wrong.
As the method is const, it should not destroy the Block's elements.

**Comment by alexey-milovidov** [comment]
It looks like the idea is incorrect.

**Comment by alexey-milovidov** [comment]
> Changelog category (leave one)

You did not leave one.

**Pull request closed without merging**